packages/venia-ui/lib/components/CategoryTree/categoryTree.js

```diff
@@ -17,33 +17,20 @@ const Tree = props => {
         updateCategories
     } = props;
 
-    const classes = mergeClasses(defaultClasses, props.classes);
-    const [queryResult, queryApi] = useQuery(MENU_QUERY);
-    const { data } = queryResult;
-    const { runQuery } = queryApi;
-
-    // fetch categories
-    useEffect(() => {
-        if (categoryId != null) {
-            runQuery({ variables: { id: categoryId } });
-        }
-    }, [categoryId, runQuery]);
-
-    // update redux with fetched categories
-    useEffect(() => {
-        if (data && data.category) {
-            updateCategories(data.category);
-        }
-    }, [data, updateCategories]);
+    const mixinProps = useCategoryTree({
+        categories,
+        categoryId,
+        query: MENU_QUERY,
```

Comment:
I think this is a good choice -- provide query via argument rather than importing within hook.

Reply:
Yeah, I think we have to let people customize the query or else these hooks will quickly lose value. That said, we'll have to find a way to set expectations for what kind of query should be passed in.